docs/dev/hacking-guide.md

```diff
@@ -0,0 +1,224 @@
+        "clusterAPIControllerAzure": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerGCP": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerOvirt": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "clusterAPIControllerVSphere": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalOperator": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalIronic": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalIronicInspector": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalIpaDownloader": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalMachineOsDownloader": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...",
+        "baremetalStaticIpManager": "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:..."
+    }
+```
+
+```
+oc edit configmap/machine-api-operator-images -n openshift-machine-api
+```
+
+With the new image information loaded into the ConfigMap, the next thing you might do is replace the Machine API operator. This operator controls how the specific cloud controllers are deployed and coordinated. You only change this component if there is something you are testing.
+
+The easiest way to change this operator is to change the image reference in the deployment. The commands you can use:
+
+```
+oc edit deploy/machine-api-operator -n openshift-machine-api
+```
+
+After changing the download reference in the images ConfigMap an easy way to swap out the controller is to let the Machine API operator do it for you. You can delete the deployment associated with the cloud provider controller and then the Machine API operator will create a new one for you, like this:
+
+```
+oc delete deploy/machine-api-controllers -n openshift-machine-api
+```
+
+## How to run e2e tests
+
+For running e2e on your cluster refer to this [document](https://github.com/openshift/cluster-api-actuator-pkg#running-the-cluster-autoscaler-operator-e2e-tests-against-an-openshift-cluster)
+
+### Running specific e2e tests
+In case you want to disable MHC, autoscaler or infra test, you can remove or comment them here - https://github.com/openshift/cluster-api-actuator-pkg/blob/master/pkg/e2e_test.go#L20
+
+Example of disabling autoscaler tests:
+```
+…
+// _ "github.com/openshift/cluster-api-actuator-pkg/pkg/autoscaler"
+_ "github.com/openshift/cluster-api-actuator-pkg/pkg/infra"
+_ "github.com/openshift/cluster-api-actuator-pkg/pkg/machinehealthcheck"
+_ "github.com/openshift/cluster-api-actuator-pkg/pkg/operators"
+)
+
+```
+
+## How to update dependencies
+machine-api-operator is vendored in every provider repository.
+
+ - Checkout cloud provider repo
+ - Run `go get github.com/openshit/machine-api-operator@master`
+ - Run `make vendor`
```

Comment:
i think we should turn this into a code block like the other examples to keep things consistent. for example

```
git checkout github.com/openshfit/$provider_repository_name
cd $provider_repository_name
go get github.com/openshit/machine-api-operator@master
make vendor
```